docs/decisions/07-posts-data-model.md

```diff
@@ -0,0 +1,162 @@
+
+For reference, the model we have now is that various classes (`Note` and `Image` at the moment) implement an `IContentRef` interface, and that's kind of it. Those objects can be assigned a resolvable URI based on their types, and those URIs are the critical component of Feed records. To actually use this system in practice, the client would retrieve a batch of records for the feed, and it would consist of little more than URI which it would have to then issue individual follow-up requests to resolve. Which is not great. It's also surprisingly difficult to map AP docs to these objects. I'm not sure this design solves that problem on its own, but I think it does become easier to build a `ContentFactory` or similar, which was likely always going to be necessary. Then we can map them to `Content` objects, as built by the factory, which can handle the logic of deciding what concrete type to construct.
+
+## Discussion
+
+### minor activities
+A Post can have more than one content, and a content item can theoretically exist in more than one post. If a post is `Liked` by someone, should that Like follow the content to other posts? I think probably not, right? For example, suppose you upload a photo, then post that photo with a note. The later you add that photo to an album. Should the view of the photo in the album show all the Likes that the Post received? What if you reuse the photo in another post in the future?
+
+### notes or just post body?
+Is it better to have notes as a content object of the post, or should the note just be sort of body properties on the post? It's probably more complicated to serialize reliably if the post body becomes a note on the wire. But it would save a join for the large majority of records. Sort of. The join is still necessary for the query, because we still need the other content items. But there would be no results from the other joined tables most of the time, and I expect the query planner to handle that efficiently.
+
+### schema
+Making each content record include-able in only one post saves on potentially a _ton_ of many-many joins. This makes it more involved to re-use uploaded media. For example, to reuse the same reaction gif repeatedly. We would need to introduce other models to track those uploads as files, not just post content. But, that seems worth it to me.
+
+We can defer that model design until we start work on the [file/media features](https://github.com/Letterbook/Letterbook/issues/139).
+
+### hypothetical compatibility
+Not attempting to be universally future compatible seems reasonable. It's not like it would be possible anyway. We can work to improve compatibility in the future, if such an incompatible peer service should ever be created.
+
+### need the interface?
+Do we need `IContent`, or is the base/abstract `Content` class enough?
```

Comment:
There's potentially some testing benefit via mocking? But I'm not sure where or why we'd need that? Why do we need an abstract `Content` though - seems like default implementations and extension methods would cover most of the needs of the general class? Maybe I am overlooking something?

Reply:
Yeah, there may not be much benefit to making it abstract. My thinking there was mostly that I don't want that base class to accidentally end up in the relational db schema.